### ISSUE TITLE
LocalFinalVariableName false-positive

### ISSUE THREAD

**Issue by boris-petrov** [checkstyle/checkstyle#8879]
cat config.xml
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LocalFinalVariableName"/>
  </module>
</module>
```

cat Test.java
```java
import java.util.concurrent.Semaphore;

public class Test {
	private static final Lock LOCK = new Lock();

	public void foo() {
		LOCK.lock();
		try (LOCK) { // violation
		}
	}
}

class Lock implements AutoCloseable {
	private final Semaphore semaphore;

	public Lock() {
		this.semaphore = new Semaphore(1);
	}

	public void lock() {
		try {
			semaphore.acquire();
		} catch (InterruptedException exception) {
			Thread.currentThread().interrupt();
		}
	}

	public void unlock() {
		semaphore.release();
	}

	@Override
	public void close() {
		unlock();
	}
}
```

java -Duser.language=en -Duser.country=US -jar checkstyle-8.36.2-all.jar -c config.xml Test.java
```
Starting audit...
[ERROR] /home/boris/Test.java:8:22: Name 'LOCK' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalFinalVariableName]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

`try (LOCK)` should be valid and no warning should be emitted.

**Comment by strkkk** [comment]
Fix is merged
